widgets/common/src/labels/LabelsInput.jsx

```diff
@@ -11,19 +11,29 @@ const iconStyle = {
 };
 
 export default function LabelsInput({ hideInitialLabels, initialLabels, onChange, toolbox }) {
-    const { useEffect } = React;
+    const { useRef, useEffect } = React;
     const {
-        Basic: { Divider, Form, Icon, Segment },
+        Basic: { Divider, Form, Icon, Popup, Segment },
         Common: { RevertToDefaultIcon },
-        Hooks: { useResettableState, useToggle },
+        Hooks: { useBoolean, useResettableState, useToggle },
         Utils: { combineClassNames },
         i18n
     } = Stage;
 
+    const [addingLabel, setAddingLabel, unsetAddingLabel] = useBoolean();
     const [labels, setLabels, resetLabels] = useResettableState(hideInitialLabels ? [] : initialLabels);
     const [open, toggleOpen] = useToggle();
     const [newLabelKey, setNewLabelKey, resetNewLabelKey] = useResettableState('');
     const [newLabelValue, setNewLabelValue, resetNewLabelValue] = useResettableState('');
+    const keyDropdownRef = useRef();
+
+    const newLabelIsProvided = !!newLabelKey && !!newLabelValue;
+    const newLabelIsAlreadyPresent = (function isNewLabelAlreadyPresent() {
+        const label = { key: newLabelKey, value: newLabelValue };
+        return !!(_.find(labels, label) || (hideInitialLabels && _.find(initialLabels, label)));
```

Comment:
Every time I read this line I need to think for a bit longer than is probably necessary :smile: This is because of how the condition is structured.

I have 2 suggestions here:
1. How about using `_.includes` instead of `_.find` here? We could then skip the `!!`
2. Optional: IMO it would be more readable to do 

    ```suggestion
            return _.includes([...labels, ...(hideInitialLabels ? initialLabels : [])], label);
    ```

Reply:
`_.includes` cannot handle objects properly (comparing references), that's why I didn't use it here.
But I implemented part of your suggestion.

Reply:
Good to know :slightly_smiling_face: I assumed `_.includes` and `_.find` work similarly

Thanks :+1: